### PR TITLE
Add `deno` compilation target

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | support `deno` as a compilation target
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2680
| Any Dependency Changes?  |
| License                  | MIT

In this PR we add `deno` compat data sourced from the [compat-table](https://github.com/kangax/compat-table). Now `preset-env` recognizes `deno` as a compilation target:
```json
{
  "presets": [["env", { "targets": { "deno": "1.0" } }]]
}
```
Unlike `node`, `deno: "current"` is not yet supported because Babel does not support `deno`.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14944"></a>

### CODE REVIEW

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by liuxingbaoyu — APPROVED** [review]

**Comment by zloirock** [comment]
What about adding Hermes that's in `compat-table` for a long time?

**Comment by zloirock** [comment]
https://github.com/babel/babel/issues/14831

**Review by existentialism — APPROVED** [review]

File: packages/babel-preset-env/test/fixtures/preset-options/deno-1_0/output.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+var _selfBrandCheck = /*#__PURE__*/new WeakSet();
+// These syntaxes should be transpiled:
+// static {} / ??= / #self in / C?.#self
+// The class declaration and the static private property should not be transpiled
+class C {
+  static #self = (_selfBrandCheck.add(this), C);
+  static #_ = C.#self ?? (C.#self = _selfBrandCheck.has(C === null || C === void 0 ? void 0 : C.#self));
```
Comment (liuxingbaoyu) [inline packages/babel-preset-env/test/fixtures/preset-options/deno-1_0/output.js:9]: I don't understand why this is, in the previous babel8 test, the optional chain was not transpiled, now it is transpiled, and the test also passes.😕

**Comment by JLHwung** [comment]
✔️ for the recent rebase. @liuxingbaoyu Thanks.

**Pull request merged**